pkg/messaging/mqtt/setup_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package mqtt_test
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+	"testing"
+	"time"
+
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+	mainflux_log "github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/pkg/messaging"
+	mqtt_pubsub "github.com/mainflux/mainflux/pkg/messaging/mqtt"
+	"github.com/ory/dockertest/v3"
+)
+
+var (
+	pubsub  messaging.PubSub
+	logger  mainflux_log.Logger
+	address string
+)
+
+const (
+	username = "mainflux-mqtt"
+	qos      = 2
+)
+
+func TestMain(m *testing.M) {
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		log.Fatalf("Could not connect to docker: %s", err)
+	}
+
+	container, err := pool.Run("eclipse-mosquitto", "1.6.13", nil)
+	if err != nil {
+		log.Fatalf("Could not start container: %s", err)
+	}
+
+	handleInterrupt(m, pool, container)
+
+	address = fmt.Sprintf("%s:%s", "localhost", container.GetPort("1883/tcp"))
+	pool.MaxWait = 120 * time.Second
```

Comment:
This is not the big remark because this is a test, but I would consider putting these magic numbers like `120` or `1883` port number in constants somewhere. Same with the Mosquitto version.

If we ever have a need to change these (and I guess this might arrive for Mosquitto version), I would prefer that these are visibly declared at the top of the file.

Reply:
Pushed these changes @drasko and @dborovcanin.
Is this format correct?